cmd/pulumi-test-language/README.md

```diff
@@ -1,63 +1,134 @@
-pulumi-language-test runs a gRPC interface that language plugins can use to run a suite of standard tests.
+(language-conformance-tests)=
+# Language conformance tests
+
+*Language conformance tests* (often just *conformance tests*) are a class of
+integration test that assert various properties that should hold true across all
+[language runtimes](language-runtimes), in essence providing a specification
+that language runtimes must conform to. They are structured as follows:
+
+* The "program under test" is expressed using [PCL](pcl). The program can
+  specify resources and functions supplied by one or more test resource
+  providers with fixed, known implementations, as well as provider-agnostic
+  entities such as stack outputs.
+* A set of assertions are made about the [state](state-snapshots) of the
+  resources, functions, outputs, etc. in the program before and after it has
+  been executed. These assertions should hold true regardless of the language
+  being used to define and execute the program.
+
+For each test run and language, then:
+
+* If the test requires one or more providers, SDKs are generated from the
+  relevant test providers, exercising SDK generation for the given language.
+* The PCL program is converted into a program in the target language, exercising
+  program generation for the given language.
+* The generated program is executed using a compiled language host running as a
+  separate process and the assertions are checked, exercising program execution
+  for the given language.
```

Comment:
Worth noting all the generated code is "snapshot" tested to ensure it doesn't change unexpectedly.